oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexCleaner.java

```diff
@@ -101,10 +103,26 @@ public void run() {
                     String remoteIndexName = ElasticIndexNameHelper.getRemoteIndexName(indexPrefix, childNodeEntry.getNodeState(), indexPath);
                     if (remoteIndexName != null) {
                         existingIndices.add(remoteIndexName);
+                    } else if (typeValue.equals(ElasticIndexDefinition.TYPE_ELASTICSEARCH)){
```

Comment:
I usually reverse the order of operands for string comparisons against a constant, so that:

    ElasticIndexDefinition.TYPE_ELASTICSEARCH.equals(typeValue)

That way, there is no risk of NPE. I see this prevented above using (typeProperty != null ? ...), but I think I would still reverse the order... that way, even refactoring the code can't result in a NPE.